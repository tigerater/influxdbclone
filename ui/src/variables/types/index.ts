export type FluxColumnType =
  | 'boolean'
  | 'unsignedLong'
  | 'long'
  | 'double'
  | 'string'
  | 'base64Binary'
  | 'dateTime'
  | 'duration'

export interface VariableValues {
  values: string[]
  valueType: FluxColumnType
  selectedValue: string
  error?: string
}

export interface VariableValuesByID {
  [variableID: string]: VariableValues
}

export interface ValueSelections {
  [variableID: string]: string
}
