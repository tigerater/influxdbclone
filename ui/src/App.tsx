// Libraries
import React, {SFC, ReactChildren} from 'react'
import {connect} from 'react-redux'

// Components
import {AppWrapper} from '@influxdata/clockface'
import Nav from 'src/pageLayout'
import TooltipPortal from 'src/portals/TooltipPortal'
import NotesPortal from 'src/portals/NotesPortal'
import Notifications from 'src/shared/components/notifications/Notifications'
import OverlayController from 'src/overlays/components/OverlayController'
import CloudNav from 'src/pageLayout/components/CloudNav'
import CloudOnly from 'src/shared/components/cloud/CloudOnly'

// Types
import {AppState} from 'src/types'

interface StateProps {
  inPresentationMode: boolean
}
interface OwnProps {
  children: ReactChildren
}

type Props = OwnProps & StateProps

const App: SFC<Props> = ({children, inPresentationMode}) => (
  <>
    <CloudOnly>
      <CloudNav />
    </CloudOnly>
    <AppWrapper presentationMode={inPresentationMode}>
      <Notifications />
      <TooltipPortal />
      <NotesPortal />
      <OverlayController />
      <Nav />
      {children}
    </AppWrapper>
  </>
)

const mstp = (state: AppState): StateProps => {
  const {
    app: {
      ephemeral: {inPresentationMode},
    },
  } = state

  return {inPresentationMode}
}

export default connect<StateProps, {}>(
  mstp,
  null
)(App)
