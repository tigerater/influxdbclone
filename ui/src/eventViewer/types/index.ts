// Types
import {ComponentType} from 'react'
import {CancelBox} from 'src/types'
import {State, Dispatch} from 'src/eventViewer/components/EventViewer.reducer'

export interface Row {
  [k: string]: any
}

export type LoadRows = (options: LoadRowsOptions) => CancelBox<Row[]>

export interface LoadRowsOptions {
  offset: number
  limit: number
  since: number
  until?: number
  filter?: SearchExpr
}

export type SearchExpr = SearchNotExpr | SearchBinaryExpr | string | number

export interface SearchNotExpr {
  type: 'NOT'
  value: SearchExpr
}

export interface SearchBinaryExpr {
  type: 'BINARY_EXPR'
  left: SearchExpr
  right: SearchExpr
  op: SearchBinaryOp
}

export type SearchBinaryOp =
  | 'AND'
  | 'OR'
  | 'EQUALS'
  | 'NOT_EQUALS'
  | 'REG_MATCH'
  | 'NOT_REG_MATCH'

export interface EventViewerChildProps {
  state: State
  dispatch: Dispatch
  loadRows: LoadRows
}

export interface FieldComponents {
  [fieldName: string]: ComponentType<{row: Row}>
}
