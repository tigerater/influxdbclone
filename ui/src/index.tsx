import 'babel-polyfill'
import 'abortcontroller-polyfill/dist/polyfill-patch-fetch'

import React, {PureComponent} from 'react'
import {render} from 'react-dom'
import {Provider} from 'react-redux'
import {Router, Route, useRouterHistory, IndexRoute} from 'react-router'
import {createHistory, History} from 'history'

import {CLOUD} from 'src/shared/constants'
import configureStore from 'src/store/configureStore'
import {loadLocalStorage} from 'src/localStorage'

import {getRootNode} from 'src/utils/nodes'
import {getBrowserBasepath} from 'src/utils/basepath'

// Components
import App from 'src/App'
import GetOrganizations from 'src/shared/containers/GetOrganizations'
import Setup from 'src/Setup'
import Signin from 'src/Signin'
import SigninPage from 'src/onboarding/containers/SigninPage'
import Logout from 'src/Logout'
import TaskPage from 'src/tasks/containers/TaskPage'
import TasksPage from 'src/tasks/containers/TasksPage'
import TaskRunsPage from 'src/tasks/components/TaskRunsPage'
import TaskEditPage from 'src/tasks/containers/TaskEditPage'
import DashboardContainer from 'src/dashboards/components/DashboardContainer'
import DashboardsIndex from 'src/dashboards/components/dashboard_index/DashboardsIndex'
import DashboardExportOverlay from 'src/dashboards/components/DashboardExportOverlay'
import DashboardImportOverlay from 'src/dashboards/components/DashboardImportOverlay'
import CreateFromTemplateOverlay from 'src/templates/components/createFromTemplateOverlay/CreateFromTemplateOverlay'
import CreateVariableOverlay from 'src/variables/components/CreateVariableOverlay'
import DataExplorerPage from 'src/dataExplorer/components/DataExplorerPage'
import SaveAsOverlay from 'src/dataExplorer/components/SaveAsOverlay'
import {MePage} from 'src/me'
import NotFound from 'src/shared/components/NotFound'
import GetLinks from 'src/shared/containers/GetLinks'
import GetMe from 'src/shared/containers/GetMe'
import UnauthenticatedApp from 'src/shared/containers/UnauthenticatedApp'
import TaskExportOverlay from 'src/tasks/components/TaskExportOverlay'
import TaskImportOverlay from 'src/tasks/components/TaskImportOverlay'
import EditVEO from 'src/dashboards/components/EditVEO'
import NewVEO from 'src/dashboards/components/NewVEO'
import OnboardingWizardPage from 'src/onboarding/containers/OnboardingWizardPage'
import BucketsIndex from 'src/buckets/containers/BucketsIndex'
import TemplatesIndex from 'src/templates/containers/TemplatesIndex'
import TelegrafsPage from 'src/telegrafs/containers/TelegrafsPage'
import ClientLibrariesPage from 'src/clientLibraries/containers/ClientLibrariesPage'
import ClientCSharpOverlay from 'src/clientLibraries/components/ClientCSharpOverlay'
import ClientGoOverlay from 'src/clientLibraries/components/ClientGoOverlay'
import ClientJavaOverlay from 'src/clientLibraries/components/ClientJavaOverlay'
import ClientJSOverlay from 'src/clientLibraries/components/ClientJSOverlay'
import ClientPythonOverlay from 'src/clientLibraries/components/ClientPythonOverlay'
import TemplateImportOverlay from 'src/templates/components/TemplateImportOverlay'
import TemplateExportOverlay from 'src/templates/components/TemplateExportOverlay'
import VariablesIndex from 'src/variables/containers/VariablesIndex'
import ScrapersIndex from 'src/scrapers/containers/ScrapersIndex'
import VariableImportOverlay from 'src/variables/components/VariableImportOverlay'
import VariableExportOverlay from 'src/variables/components/VariableExportOverlay'
import SetOrg from 'src/shared/containers/SetOrg'
import RouteToOrg from 'src/shared/containers/RouteToOrg'
import CreateOrgOverlay from 'src/organizations/components/CreateOrgOverlay'
import CreateScraperOverlay from 'src/scrapers/components/CreateScraperOverlay'
import TokensIndex from 'src/authorizations/containers/TokensIndex'
import MembersIndex from 'src/members/containers/MembersIndex'
import LabelsIndex from 'src/labels/containers/LabelsIndex'
import TemplateViewOverlay from 'src/templates/components/TemplateViewOverlay'
import LineProtocolWizard from 'src/dataLoaders/components/lineProtocolWizard/LineProtocolWizard'
import CollectorsWizard from 'src/dataLoaders/components/collectorsWizard/CollectorsWizard'
import TelegrafInstructionsOverlay from 'src/telegrafs/components/TelegrafInstructionsOverlay'
import OrgProfilePage from 'src/organizations/containers/OrgProfilePage'
import RenameOrgOverlay from 'src/organizations/components/RenameOrgOverlay'
import UpdateBucketOverlay from 'src/buckets/components/UpdateBucketOverlay'
import RenameBucketOverlay from 'src/buckets/components/RenameBucketOverlay'
import RenameVariableOverlay from 'src/variables/components/RenameVariableOverlay'
import UpdateVariableOverlay from 'src/variables/components/UpdateVariableOverlay'
import TaskImportFromTemplateOverlay from 'src/tasks/components/TaskImportFromTemplateOverlay'
import StaticTemplateViewOverlay from 'src/templates/components/StaticTemplateViewOverlay'
import CheckHistory from 'src/alerting/components/CheckHistory'
import AlertingIndex from 'src/alerting/components/AlertingIndex'
import AlertHistoryIndex from 'src/alerting/components/AlertHistoryIndex'
import BucketsDeleteDataOverlay from 'src/shared/components/DeleteDataOverlay'
import DEDeleteDataOverlay from 'src/dataExplorer/components/DeleteDataOverlay'
import NewThresholdCheckEO from 'src/alerting/components/NewThresholdCheckEO'
import NewDeadmanCheckEO from 'src/alerting/components/NewDeadmanCheckEO'
import EditCheckEO from 'src/alerting/components/EditCheckEO'
import NewRuleOverlay from 'src/alerting/components/notifications/NewRuleOverlay'
import EditRuleOverlay from 'src/alerting/components/notifications/EditRuleOverlay'
import NewEndpointOverlay from 'src/alerting/components/endpoints/NewEndpointOverlay'
import EditEndpointOverlay from 'src/alerting/components/endpoints/EditEndpointOverlay'
import NoOrgsPage from 'src/organizations/containers/NoOrgsPage'

// Overlays
import OverlayHandler, {
  RouteOverlay,
} from 'src/overlays/components/RouteOverlay'
const AddNoteOverlay = RouteOverlay(OverlayHandler, 'add-note', router => {
  router.push(
    `/orgs/${router.params.orgID}/dashboards/${router.params.dashboardID}`
  )
})
const EditNoteOverlay = RouteOverlay(OverlayHandler, 'edit-note', router => {
  router.push(
    `/orgs/${router.params.orgID}/dashboards/${router.params.dashboardID}`
  )
})
const AllAccessTokenOverlay = RouteOverlay(
  OverlayHandler,
  'add-master-token',
  router => {
    router.push(`/orgs/${router.params.orgID}/load-data/tokens`)
  }
)
const BucketsTokenOverlay = RouteOverlay(
  OverlayHandler,
  'add-token',
  router => {
    router.push(`/orgs/${router.params.orgID}/load-data/tokens`)
  }
)
const TelegrafConfigOverlay = RouteOverlay(
  OverlayHandler,
  'telegraf-config',
  router => {
    router.push(`/orgs/${router.params.orgID}/load-data/telegrafs`)
  }
)
const TelegrafOutputOverlay = RouteOverlay(
  OverlayHandler,
  'telegraf-output',
  router => {
    router.push(`/orgs/${router.params.orgID}/load-data/telegrafs`)
  }
)

// Actions
import {disablePresentationMode} from 'src/shared/actions/app'

// Styles
import 'src/style/chronograf.scss'
import '@influxdata/clockface/dist/index.css'

const rootNode = getRootNode()
const basepath = getBrowserBasepath()

declare global {
  interface Window {
    basepath: string
  }
}

// Older method used for pre-IE 11 compatibility
window.basepath = basepath

const history: History = useRouterHistory(createHistory)({
  basename: basepath, // this is written in when available by the URL prefixer middleware
})

export const store = configureStore(loadLocalStorage(), history)
const {dispatch} = store

if (window['Cypress']) {
  window['store'] = store
}

history.listen(() => {
  dispatch(disablePresentationMode())
})

window.addEventListener('keyup', event => {
  const escapeKeyCode = 27
  // fallback for browsers that don't support event.key
  if (event.key === 'Escape' || event.keyCode === escapeKeyCode) {
    dispatch(disablePresentationMode())
  }
})

class Root extends PureComponent {
  public render() {
    return (
      <Provider store={store}>
        <Router history={history}>
          <Route component={GetLinks}>
            <Route component={Setup}>
              <Route path="/onboarding">
                <Route path=":stepID" component={OnboardingWizardPage} />
                <Route
                  path=":stepID/:substepID"
                  component={OnboardingWizardPage}
                />
                <Route component={UnauthenticatedApp}>
                  <Route path="/signin" component={SigninPage} />
                  <Route path="/logout" component={Logout} />
                </Route>
              </Route>
              <Route component={Signin}>
                <Route component={GetMe}>
                  <Route component={GetOrganizations}>
                    <Route path="/">
                      <Route path="no-orgs" component={NoOrgsPage} />
                      <IndexRoute component={RouteToOrg} />
                      <Route path="orgs" component={App}>
                        <Route path="new" component={CreateOrgOverlay} />
                        <Route path=":orgID" component={SetOrg}>
                          <IndexRoute component={MePage} />
                          <Route path="tasks" component={TasksPage}>
                            <Route
                              path=":id/export"
                              component={TaskExportOverlay}
                            />
                            <Route
                              path="import"
                              component={TaskImportOverlay}
                            />
                            <Route
                              path="import/template"
                              component={TaskImportFromTemplateOverlay}
                            />
                          </Route>
                          <Route
                            path="tasks/:id/runs"
                            component={TaskRunsPage}
                          />
                          <Route path="tasks/new" component={TaskPage} />
                          <Route path="tasks/:id" component={TaskEditPage} />
                          <Route
                            path="data-explorer"
                            component={DataExplorerPage}
                          >
                            <Route path="save" component={SaveAsOverlay} />
                            <Route
                              path="delete-data"
                              component={DEDeleteDataOverlay}
                            />
                          </Route>
                          <Route path="dashboards" component={DashboardsIndex}>
                            <Route
                              path="import"
                              component={DashboardImportOverlay}
                            />
                            <Route
                              path="import/template"
                              component={CreateFromTemplateOverlay}
                            />
                            <Route
                              path=":dashboardID/export"
                              component={DashboardExportOverlay}
                            />
                          </Route>
                          <Route
                            path="dashboards/:dashboardID"
                            component={DashboardContainer}
                          >
                            <Route path="cells">
                              <Route path="new" component={NewVEO} />
                              <Route path=":cellID/edit" component={EditVEO} />
                            </Route>
                            <Route path="notes">
                              <Route path="new" component={AddNoteOverlay} />
                              <Route
                                path=":cellID/edit"
                                component={EditNoteOverlay}
                              />
                            </Route>
                          </Route>
                          <Route path="me" component={MePage} />
                          <Route path="load-data">
                            <IndexRoute component={BucketsIndex} />
                            <Route path="tokens" component={TokensIndex}>
                              <Route path="generate">
                                <Route
                                  path="all-access"
                                  component={AllAccessTokenOverlay}
                                />
                                <Route
                                  path="buckets"
                                  component={BucketsTokenOverlay}
                                />
                              </Route>
                            </Route>
                            <Route path="buckets" component={BucketsIndex}>
                              <Route path=":bucketID">
                                <Route
                                  path="line-protocols/new"
                                  component={LineProtocolWizard}
                                />
                                <Route
                                  path="telegrafs/new"
                                  component={CollectorsWizard}
                                />
                                <Route
                                  path="scrapers/new"
                                  component={CreateScraperOverlay}
                                />
                                <Route
                                  path="edit"
                                  component={UpdateBucketOverlay}
                                />
                                <Route
                                  path="delete-data"
                                  component={BucketsDeleteDataOverlay}
                                />
                                <Route
                                  path="rename"
                                  component={RenameBucketOverlay}
                                />
                              </Route>
                            </Route>
                            <Route path="telegrafs" component={TelegrafsPage}>
                              <Route
                                path=":id/view"
                                component={TelegrafConfigOverlay}
                              />
                              <Route
                                path=":id/instructions"
                                component={TelegrafInstructionsOverlay}
                              />
                              <Route
                                path="output"
                                component={TelegrafOutputOverlay}
                              />
                              <Route path="new" component={CollectorsWizard} />
                            </Route>
                            <Route path="scrapers" component={ScrapersIndex}>
                              <Route
                                path="new"
                                component={CreateScraperOverlay}
                              />
                            </Route>
                            <Route
                              path="client-libraries"
                              component={ClientLibrariesPage}
                            >
                              <Route
                                path="csharp"
                                component={ClientCSharpOverlay}
                              />
                              <Route path="go" component={ClientGoOverlay} />
                              <Route
                                path="java"
                                component={ClientJavaOverlay}
                              />
                              <Route
                                path="javascript-node"
                                component={ClientJSOverlay}
                              />
                              <Route
                                path="python"
                                component={ClientPythonOverlay}
                              />
                            </Route>
                          </Route>
                          <Route path="settings">
                            {CLOUD ? (
                              <IndexRoute component={VariablesIndex} />
                            ) : (
                              <>
                                <IndexRoute component={MembersIndex} />
                                <Route
                                  path="members"
                                  component={MembersIndex}
                                />
                              </>
                            )}

                            <Route path="templates" component={TemplatesIndex}>
                              <Route
                                path="import"
                                component={TemplateImportOverlay}
                              />
                              <Route
                                path=":id/export"
                                component={TemplateExportOverlay}
                              />
                              <Route
                                path=":id/view"
                                component={TemplateViewOverlay}
                              />
                              <Route
                                path=":id/static/view"
                                component={StaticTemplateViewOverlay}
                              />
                            </Route>
                            <Route path="variables" component={VariablesIndex}>
                              <Route
                                path="import"
                                component={VariableImportOverlay}
                              />
                              <Route
                                path=":id/export"
                                component={VariableExportOverlay}
                              />
                              <Route
                                path="new"
                                component={CreateVariableOverlay}
                              />
                              <Route
                                path=":id/rename"
                                component={RenameVariableOverlay}
                              />
                              <Route
                                path=":id/edit"
                                component={UpdateVariableOverlay}
                              />
                            </Route>
                            <Route path="labels" component={LabelsIndex} />
                            <Route path="profile" component={OrgProfilePage}>
                              <Route
                                path="rename"
                                component={RenameOrgOverlay}
                              />
                            </Route>
                          </Route>
                          <Route path="alerting" component={AlertingIndex}>
                            <Route
                              path="checks/new-threshold"
                              component={NewThresholdCheckEO}
                            />
                            <Route
                              path="checks/new-deadman"
                              component={NewDeadmanCheckEO}
                            />
                            <Route
                              path="checks/:checkID/edit"
                              component={EditCheckEO}
                            />
                            <Route
                              path="rules/new"
                              component={NewRuleOverlay}
                            />
                            <Route
                              path="rules/:ruleID/edit"
                              component={EditRuleOverlay}
                            />
                            <Route
                              path="endpoints/new"
                              component={NewEndpointOverlay}
                            />
                            <Route
                              path="endpoints/:endpointID/edit"
                              component={EditEndpointOverlay}
                            />
                          </Route>
                          <Route
                            path="alert-history"
                            component={AlertHistoryIndex}
                          />
                          <Route
                            path="checks/:checkID"
                            component={CheckHistory}
                          />
                        </Route>
                      </Route>
                    </Route>
                  </Route>
                </Route>
              </Route>
            </Route>
          </Route>
          <Route path="*" component={NotFound} />
        </Router>
      </Provider>
    )
  }
}

if (rootNode) {
  render(<Root />, rootNode)
}
