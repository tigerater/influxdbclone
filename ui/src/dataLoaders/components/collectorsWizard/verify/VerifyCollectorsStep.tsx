// Libraries
import React, {PureComponent} from 'react'
import {connect} from 'react-redux'
import _ from 'lodash'

// Components
import {Form} from '@influxdata/clockface'
import DataStreaming from 'src/dataLoaders/components/verifyStep/DataStreaming'
import OnboardingButtons from 'src/onboarding/components/OnboardingButtons'
import FancyScrollbar from 'src/shared/components/fancy_scrollbar/FancyScrollbar'

// Types
import {CollectorsStepProps} from 'src/dataLoaders/components/collectorsWizard/CollectorsWizard'
import {AppState} from 'src/types'

// Decorators
import {ErrorHandling} from 'src/shared/decorators/errors'

type OwnProps = CollectorsStepProps

interface StateProps {
  username: string
  telegrafConfigID: string
  bucket: string
  org: string
  token: string
}

export type Props = StateProps & OwnProps

@ErrorHandling
export class VerifyCollectorStep extends PureComponent<Props> {
  public render() {
    const {
      telegrafConfigID,
      bucket,
      org,
      onDecrementCurrentStepIndex,
      onExit,
      token,
    } = this.props

    return (
      <Form onSubmit={onExit} className="data-loading--form">
        <FancyScrollbar
          autoHide={false}
          className="data-loading--scroll-content"
        >
          <div>
            <h3 className="wizard-step--title">Test your Configuration</h3>
            <h5 className="wizard-step--sub-title">
              Start Telegraf and ensure data is being written to InfluxDB
            </h5>
          </div>
          <DataStreaming
            org={org}
            bucket={bucket}
            token={token}
            configID={telegrafConfigID}
          />
        </FancyScrollbar>
        <OnboardingButtons
          onClickBack={onDecrementCurrentStepIndex}
          nextButtonText="Finish"
          className="data-loading--button-container"
        />
      </Form>
    )
  }
}

const mstp = ({
  dataLoading: {
    dataLoaders: {telegrafConfigID, token},
    steps: {bucket, org},
  },
  me: {name},
}: AppState): StateProps => ({
  username: name,
  telegrafConfigID,
  bucket,
  org,
  token,
})

export default connect<StateProps, {}, OwnProps>(mstp)(VerifyCollectorStep)
