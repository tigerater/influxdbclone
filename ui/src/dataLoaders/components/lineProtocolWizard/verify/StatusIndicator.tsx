// Libraries
import React, {PureComponent} from 'react'
import classnames from 'classnames'
import {connect} from 'react-redux'

// Components
import {SparkleSpinner} from '@influxdata/clockface'

// Types
import {RemoteDataState} from 'src/types'
import {AppState} from 'src/types'

interface StateProps {
  status: RemoteDataState
  errorMessage: string
}

type Props = StateProps

export class StatusIndicator extends PureComponent<Props> {
  public render() {
    const {status} = this.props
    return (
      <>
        <div className="wizard-step--top-container">
          <div className="wizard-step--sparkle-container">
            <SparkleSpinner loading={status} />
          </div>
        </div>
        <div className="wizard-step--footer">
          <div className={this.footerClass}>
            {this.footerText}
            {this.errorMessage}
          </div>
        </div>
        <br />
      </>
    )
  }

  private get footerClass(): string {
    const {status} = this.props

    return classnames(`wizard-step--text-state`, {
      loading: status === RemoteDataState.Loading,
      success: status === RemoteDataState.Done,
      error: status === RemoteDataState.Error,
    })
  }

  private get footerText(): string {
    switch (this.props.status) {
      case RemoteDataState.Loading:
        return 'Loading...'
      case RemoteDataState.Done:
        return 'Data Written Successfully!'
      case RemoteDataState.Error:
        return 'Unable to Write Data'
    }
  }

  private get errorMessage(): JSX.Element {
    if (this.props.status === RemoteDataState.Error)
      return (
        <>
          <br />
          Error: {this.props.errorMessage}
        </>
      )
  }
}

const mstp = ({
  dataLoading: {
    dataLoaders: {lpStatus, lpError},
  },
}: AppState): StateProps => ({
  status: lpStatus,
  errorMessage: lpError,
})

export default connect<StateProps, {}, {}>(
  mstp,
  null
)(StatusIndicator)
