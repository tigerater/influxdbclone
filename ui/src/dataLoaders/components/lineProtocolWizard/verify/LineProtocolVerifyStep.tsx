// Libraries
import React, {PureComponent} from 'react'
import _ from 'lodash'

// Components
import {Form, Overlay} from '@influxdata/clockface'
import StatusIndicator from 'src/dataLoaders/components/lineProtocolWizard/verify/StatusIndicator'
import OnboardingButtons from 'src/onboarding/components/OnboardingButtons'
import LineProtocolHelperText from 'src/dataLoaders/components/lineProtocolWizard/LineProtocolHelperText'

// Types
import {LineProtocolStepProps} from 'src/dataLoaders/components/lineProtocolWizard/LineProtocolWizard'

// Decorators
import {ErrorHandling} from 'src/shared/decorators/errors'

type Props = LineProtocolStepProps

@ErrorHandling
export class VerifyLineProtocolStep extends PureComponent<Props> {
  public render() {
    const {onDecrementCurrentStepIndex, onExit} = this.props

    return (
      <Form onSubmit={onExit}>
        <Overlay.Body style={{textAlign: 'center'}}>
          <StatusIndicator />
          <LineProtocolHelperText />
        </Overlay.Body>
        <OnboardingButtons
          onClickBack={onDecrementCurrentStepIndex}
          nextButtonText="Finish"
        />
      </Form>
    )
  }
}

export default VerifyLineProtocolStep
