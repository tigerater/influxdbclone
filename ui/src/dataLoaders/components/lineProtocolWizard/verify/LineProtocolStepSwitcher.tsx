// Libraries
import React, {PureComponent} from 'react'

// Components
import {ErrorHandling} from 'src/shared/decorators/errors'
import LineProtocol from 'src/dataLoaders/components/lineProtocolWizard/configure/LineProtocol'
import LineProtocolVerifyStep from 'src/dataLoaders/components/lineProtocolWizard/verify/LineProtocolVerifyStep'

// Types
import {LineProtocolStepProps} from 'src/dataLoaders/components/lineProtocolWizard/LineProtocolWizard'
import {Bucket, LineProtocolStep} from 'src/types'

interface Props {
  stepProps: LineProtocolStepProps
  buckets: Bucket[]
}

@ErrorHandling
class StepSwitcher extends PureComponent<Props> {
  public render() {
    const {stepProps} = this.props

    switch (stepProps.currentStepIndex) {
      case LineProtocolStep.Configure:
        return <LineProtocol {...stepProps} />
      case LineProtocolStep.Verify:
        return <LineProtocolVerifyStep {...stepProps} />
      default:
        return <div />
    }
  }
}

export default StepSwitcher
