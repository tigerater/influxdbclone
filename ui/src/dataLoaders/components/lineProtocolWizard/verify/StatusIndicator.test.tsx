// Libraries
import React from 'react'
import {shallow} from 'enzyme'

// Components
import {StatusIndicator} from 'src/dataLoaders/components/lineProtocolWizard/verify/StatusIndicator'

// Types
import {RemoteDataState} from 'src/types'

const setup = (override?) => {
  const props = {
    status: RemoteDataState.NotStarted,
    ...override,
  }

  const wrapper = shallow(<StatusIndicator {...props} />)

  return {wrapper}
}

describe('StatusIndicator', () => {
  describe('rendering', () => {
    it('renders!', () => {
      const {wrapper} = setup()
      expect(wrapper.exists()).toBe(true)
    })
  })
})
