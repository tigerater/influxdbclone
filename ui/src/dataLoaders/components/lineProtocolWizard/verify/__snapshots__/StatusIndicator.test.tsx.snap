// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatusIndicator rendering renders! 1`] = `
<Fragment>
  <div
    className="wizard-step--top-container"
  >
    <div
      className="wizard-step--sparkle-container"
    >
      <SparkleSpinner
        loading="NotStarted"
        sizePixels={100}
        testID="sparkle-spinner"
      />
    </div>
  </div>
  <div
    className="wizard-step--footer"
  >
    <div
      className="wizard-step--text-state"
    />
  </div>
  <br />
</Fragment>
`;
